addons/misra.py

```diff
@@ -496,6 +496,8 @@ def getEssentialTypeCategory(expr):
         return expr.valueType.sign
 
     if expr.valueType:
+        if expr.valueType.type == 'char':
```

Comment:
I am not sure about this fix.

I think the expected behavior is:

     (char)'1'  => no warning
     (unsigned char)'1'  => warning
     (signed char)'1'  => warning